framework/source/class/qx/util/DynamicScriptLoader.js

```diff
@@ -0,0 +1,258 @@
+ *
+ * Usage example:
+ *
+ * <code>
+ *  ... assets ...
+ * /**
+ *  * @asset(myapp/jquery/*)
+ *  * @asset(myapp/highcharts/*)
+ *  *
+ *  * @ignore(jQuery.*)
+ *  * @ignore(Highcharts.*)
+ *  ...
+ *
+ *
+ *    // in debug mode load the uncompressed unobfuscated scripts
+ *    var src = '';
+ *    var min = '.min';
+ *    if (qx.core.Environment.get("qx.debug")) {
+ *      src = '.src';
+ *      min = '';
+ *    }
+ *
+ *    // initialize the script loading
+ *    var dynLoader = new qx.util.DynamicScriptLoader([
+ *        "myapp/jquery/jquery"+min+".js",
+ *        "myapp/highcharts/highcharts"+src+".js",
+ *        "myapp/highcharts/highcharts-more"+src+".js",
+ *        "myapp/highcharts/highcharts-modifications.js"
+ *    ]);
+ *
+ *
+ *    dynLoader.addListenerOnce('ready',function(e){
+ *      console.log("all scripts have been loaded!");
+ *    });
+ *
+ *    dynLoader.addListener('failed',function(e){
+ *      var data = e.getData();
+ *      console.log("failed to load "+data.script);
+ *    });
+ *
+ *    dynLoader.start();
+ *    
+ * </code>
+ */
+qx.Class.define("qx.util.DynamicScriptLoader", {
+  extend: qx.core.Object,
+
+  /**
+   * Create a loader for the given scripts.
+   *
+   * @param scriptArr {Array|String} the uri name(s) of the script(s) to load 
+   */
+
+  construct: function (scriptArr) {
+    this.base(arguments);
+    this.__QUEUE = (qx.lang.Type.isString(scriptArr)
+                ? [ scriptArr ]
+                : qx.lang.Array.clone(scriptArr));
+  },
+
+  /*
+  *****************************************************************************
+     EVENTS
+  *****************************************************************************
+  */
+
+  events: {
+    /**
+     * fired when a script is loaded successfully. The data contains 'script' and 'status' keys.
+     */
+    loaded: 'qx.event.type.Data',
+
+    /**
+     * fired when a specific script fails loading.  The data contains 'script' and 'status' keys.
+     */
+    failed: 'qx.event.type.Data',
+
+    /**
+     * fired when all given scripts are loaded, each time loadScriptsDynamic is called.
+     */
+    ready: 'qx.event.type.Event'
+  },
+
+  statics: {
+    /**
+     * Map of scripts being added at the present time. Key is script name; value is instance of this class which
+     * is loading it.
+     */
+    __IN_PROGRESS: {},
+     /**
+      * Map of scripts that have fully loaded. Key is script name; value is true
+      */
+    __LOADED: {}
+  },
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
+
+
+  members: {
+
+    /**
+     * Array of the scripts to be loaded
+     */
+    __QUEUE: null,
+
+    /**
+     * Start loading scripts. This may only be called once!
+     *
+     */
+    start: function() {
+      if (this.isDisposed()) {
+        return;
+      }
+      this.start = function(){
```

Comment:
please can you stored a `this.__started` flag and test for it instead of changing the `start` function; although the started state cannot be turned off, it's unexpected behaviour for a declared method to suddenly point to a non-method.